lib/src/config/config.rs

```diff
@@ -119,7 +119,11 @@ impl Config {
                 return Err(self.bad_type(name, val, "an unsigned integer"));
             }
 
-            self.port = port as usize;
+            if port > (u16::max_value() as i64) {
```

Comment:
I think you should just be able to do 

```
if !(port <= u16::max_value())
```

for logic equivalence. That way you don't need to cast to an `i64`. Also with the declaration of `port` if you want it as a `u16` do

```
let port: u16 = parse!(self, name, val, as_integer, "an integer")?;
```

Which should allow you to do:

```
self.port = port;
```